### ISSUE TITLE
blocknotify does not work with RichList enabled

### ISSUE THREAD

**Issue by psipherious** [Nexusoft/Nexus#73]
by default: blocknotify works perfectly when running Nexus as CLI on Ubuntu 16.04

If richlist=1 is enabled (either in config file or via command line flag), blocknofity no longer works, it simply ceases to execute whatever command is specified while richlist is on.

Have tested blocknotify using nexus.conf and also command flag, both with richlist on and richlist off.

**Comment by VidereLicet** [comment]
This still an issue?

**Comment by psipherious** [comment]
Yes sir it does appear to be even in 2.4.5 - blocknoitfy works fine on any node not running richlist=1 but not at all on richlist nodes.

I've tried to investigate the problem myself but my C/C++ skills are limited and I have not had success so far.

**Comment by VidereLicet** [comment]
Did this get introduced through any new versions?

**Comment by psipherious** [comment]
I only started trying to use blocknotify for the first time maybe sometime around version 2.4.0 - it took me a while to figure out that richlist=1 was the *only* reason why it wasn't working on some of my nodes while it was working fine on others. I've tested it on both Oracle and LLD versions and the result is the same, anytime richlist is enabled, blocknotify doesn't do anything.

I just have a basic sh script that writes a little note to the hard drive saying that blocknotify was successful and have it setup exactly the same way on all my nodes and on the richlist ones it just does nothing.

So I can't say for sure if it used to work on older nodes. You never used to require a richlist=1 flag in older ones in order to pull the richlist anyways - it used to only work on Oracle nodes in the past and I just always used that for my richlist.
